src/or/connection.c

```diff
@@ -2658,16 +2687,27 @@ retry_listener_ports(smartlist_t *old_conns,
           break;
         }
       } else {
-        int port_matches;
-        if (wanted->port == CFG_AUTO_PORT) {
```

Comment:
What happens to this CFG_AUTO_PORT check?  it seems to get removed here, but not added somewhere else?

Reply:
Oops. Fixed in 6975f6d8cb2724f7039938350497d8ec3f36ecc5.